drivers/syscon/Kconfig

```diff
@@ -3,33 +3,31 @@
 # Copyright (c) 2021 Carlo Caione <ccaione@baylibre.com>
 # SPDX-License-Identifier: Apache-2.0
 
-#
-# SYSCON options
-#
-menuconfig SYSCON
-	bool "SYSCON (System Controller) drivers"
+# SYSCON (System Controller) drivers. System controller node represents
+# a register region containing a set of miscellaneous registers. The
+# registers are not cohesive enough to represent as any specific type
+# of device. The typical use-case is for some other node's driver, or
+# platform-specific code, to acquire a reference to the syscon node and
+# extract information from there.
```

Comment:
Why is this removed from the `help` text.

This means it's no longer visible inside `menuconfig`.

Reply:
This loops me back to my original change for this PR... where all I did was set the default value of SYSCON to `y` and kept the menu the same. That kept the help description the same and the same menu structure. If you'd like me to go back to the original 1 line change I'd be very happy to do that. It made the most sense, but you can see from the comments there was a lot of opposition to it.

Reply:
> but you can see from the comments there was a lot of opposition to it.

I don't see opposition to keep the old structure, quite the opposite, like this comment:
https://github.com/zephyrproject-rtos/zephyr/pull/38762#issuecomment-926546586
> I think we need to keep the "top-level" Kconfig for enabling SYSCON support and then have the drivers underneath that (as it is today).

as I read the comments, the opposition was on how `SYSCON` was being enabled, as it goes against how other drivers do things.

Also make sure you're not re-introducing this: https://github.com/zephyrproject-rtos/zephyr/issues/38403
that is already now a mess to clean up.

Reply:
OK, I reverted back to the original change + a change to address what you mentioned and avoid creating a `zephyr_library` unless the right Kconfig was enabled. The opposition was that this was messy and should be explicitly enabled. My take is that developers already need to add a devicetree node and set `status = "okay"`. This is explicit enough. Having `CONFIG_SYSCON=y` by default doesn't affect build time/size at all.